python/tank/descriptor/io_descriptor/base.py

```diff
@@ -524,18 +539,48 @@ def get_path(self):
 
         return None
 
+    def clone_cache(self, cache_root):
+        """
+        The descriptor system maintains an internal cache where it downloads
+        the payload that is associated with the descriptor. Toolkit supports
+        complex cache setups, where you can specify a series of path where toolkit
+        should go and look for cached items.
 
-    ###############################################################################################
-    # stuff typically implemented by deriving classes
+        This is an advanced method that helps in cases where a user wishes to
+        administer such a setup, allowing a cached payload to be copied from
+        its current location into a new cache structure.
 
-    def _get_cache_paths(self):
+        If the descriptor doesn't exist on disk yet, it will be downloaded.
+
+        :param cache_root: Root point of the cache location to copy to.
         """
-        Get a list of resolved paths, starting with the primary and
-        continuing with alternative locations where it may reside
+        # compute new location
+        new_cache_path = self._get_bundle_cache_path(cache_root)
+        log.debug("Clone cache for %r: Copying to '%s'" % (self, new_cache_path))
 
-        Note: This method only computes paths and does not perform any I/O ops.
+        # make sure we have something to copy
+        self.ensure_local()
 
-        :return: List of path strings
+        # check that we aren't trying to copy onto ourself
+        if new_cache_path == self.get_path():
```

Comment:
Technically, this should probably be before `ensure_local` since it is an error case and should result in a no-op.